pkg/pillar/priorityQueue/priorityQueue.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2021 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Queue to work with objects with sorting by priority of type
+
+package priorityQueue
+
+import (
+	"sync"
+	"time"
+)
+
+type queueObject struct {
+	lastTime time.Time
+	obj      interface{}
+}
+
+//PriorityQueue contains objects with particular type and key
+type PriorityQueue struct {
+	objs                   map[interface{}]map[string]queueObject //map[type]map[key]queueObject
+	count                  uint32
+	priorityCheckFunctions []typePriorityCheck
+	lock                   sync.Mutex
+}
+
+//typePriorityCheck returns true in case of find objType with high priority
+type typePriorityCheck func(objType interface{}) bool
+
+//InitQueue returns new PriorityQueue with defined priorityCheckFunctions
+func InitQueue(priorityCheckFunctions ...typePriorityCheck) *PriorityQueue {
+	//append with return first
+	priorityCheckFunctions = append(priorityCheckFunctions, func(obj interface{}) bool {
+		return true
```

Comment:
It took me a way too long time to figure this out; that there is a default function at the end of the list which always returns true. That could definitely be made more clear.
But also, don't we need to enable more than two priorities so we can have e.g., appinstance > device > netinst, volumes > contenttree > blob?

Reply:
You can assign several functions with object types for this. The last one will fire if others was not lucky to find types in map

Reply:
Ah - I missed that the call had a ...
But should still comment that there is default/sentinel function added and the user doesn't need to do that.